src/spikeinterface/sorters/external/kilosort4.py

```diff
@@ -205,40 +205,69 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
         # NOTE: Also modifies settings in-place
         data_dir = ""
         results_dir = sorter_output_folder
-        filename, data_dir, results_dir, probe = set_files(settings, filename, probe, probe_name, data_dir, results_dir)
+
+        filename, data_dir, results_dir, probe = set_files(
+            settings=settings,
+            filename=filename,
+            probe=probe,
+            probe_name=probe_name,
+            data_dir=data_dir,
+            results_dir=results_dir,
+        )
+
         if version.parse(cls.get_sorter_version()) >= version.parse("4.0.12"):
-            ops = initialize_ops(settings, probe, recording.get_dtype(), do_CAR, invert_sign, device, False)
+            # TODO: save_preprocesed_copy added
+            ops = initialize_ops(
+                settings=settings,
+                probe=probe,
+                data_dtype=recording.get_dtype(),
+                do_CAR=do_CAR,
```

Comment:
Ported from #3073 

> Chris
> So, at the moment, if a user specifies `skip_kilosort_preprocessing = True`, then `do_CAR` is still `True` by default and is still applied. I guess: is this the expected behaviour?
> 
> And then what do we want to do if someone explicitly specifies `skip_kilosort_preprocessing = True` and `do_CAR=True` ?

> So, at the moment, if a user specifies `skip_kilosort_preprocessing = True`, then `do_CAR` is still `True` by default and is still applied. I guess: is this the expected behaviour?
> 
> And then what do we want to do if someone explicitly specifies `skip_kilosort_preprocessing = True` and `do_CAR=True` ?

>Alessio
> So, at the moment, if a user specifies `skip_kilosort_preprocessing = True`, then `do_CAR` is still `True` by default and is still applied. I guess: is this the expected behaviour?

>Chris
> And then what do we want to do if someone explicitly specifies `skip_kilosort_preprocessing = True` and `do_CAR=True` ?

Reply:
I think because `do_CAR` is the default `True` it is not possible to determine if the user wanted to apply `CAR` or just left the default on. In this case I think most users would expect `skip_kilosort_preprocessing` to skip all preprocessing and would be surprised if CAR was still performed by default. I would suggest to raise an error if `skip_kilosort_preprocessing=False` and `do_CAR=True`. It will be annoying for users first time because by default it will raise an error, but at least it is an easy fix and super explicit. If we get issues raised from people who want to skip KS preprocessing but keep CAR (I would imagine this is very small % of users) we could reconsider.

Further on this, I believe that the KS drift correction is performed in the [code block](https://github.com/MouseLand/Kilosort/blob/c4ed5dd093e6b979756f2e5c85a781ff100079b3/kilosort/io.py#L726) that also performs whitening. So when whitening is turned off as part of `skip_kilosort_preprocessing` then drift correct is also turned off. This makes it even weirder for `skip_kilosort_preprocessing` to allow only CAR. Also (if this is correct) this behaviour can be added to the dosctring and I'd suggest raise an error if `do_correction=True` while `skip_kilosort_preprocessing=True`. Even though it will raise a lot of errors by default when trying to use `skip_kilosort_preprocessing` I feel it is quite a complex / confusing case and so best to be super explicit.